src/App.jsx

```diff
@@ -14,70 +16,58 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      {selectedGood ? (
+        <h1 className="title is-flex is-align-items-center">
+          {`${selectedGood} is selected`}
+          <button
+            onClick={() => setSelectedGood('')}
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+          />
+        </h1>
+      ) : (
+        <h1 className="title">No goods selected</h1>
+      )}
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const goodIsSelected = good === selectedGood;
+            const setGood = () => setSelectedGood(goodIsSelected ? '' : good);
```

Comment:
better to move 'const setGood = () => setSelectedGood(goodIsSelected ? '' : good);' from template to the component, because is't bad practice to make a lot of logic in a template

Reply:
Write everything inside the App (don't create additional components). 

As it is mentioned in the task I decided to move this logic to a separate function.